### DISCUSSION TITLE
Code formatting and coloring in LSP

### DISCUSSION THREAD

**Discussion by praveen69sk** [eclipse-theia/theia#12237] [Q&A]
How to add coloring and code formatting in LSP editor as we are using LSP vscode extension in theia for our custom language.
Any suggestions will be much appreciated. Thanks in advance.

**Comment by msujew** [comment]
Hey @praveen69sk,

this is quite unrelated to Theia. Language services such as syntactical/semantic highlighting and formatting need to be implemented by the vscode extension/the language server. See also the [lsp formatting](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_formatting), [lsp semantic highlight](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_semanticTokens) and [textMate grammars](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#textmate-grammars) documentation.
